### ISSUE TITLE
[Bug]: Dashboard send 15mb of JavaScript

### ISSUE THREAD

**Issue by heyarne** [saleor/saleor-dashboard#4183]
### Description of the issue

I have noticed that the current version of the Dashboard sends roughly 15MB of JavaScript. This seems to be in part because there's no minification (and no tree-shaking?) done in the build process.

### Steps to reproduce the problem

1. Deploy the current version of the saleor-dashboard
2. Open the network tab, do a hard refresh of the page

### What did you expect to happen?

I'd expect the amount of JavaScript to be siginificantly less.

### Additional information

As a quick fix added some lines to the nginx file to enable gzip compression:

```diff
--- nginx/default.conf	2022-07-05 13:15:32.585406677 +0200
+++ nginx/default.conf	2023-09-11 11:00:58.583704469 +0200
@@ -3,6 +3,15 @@
     server_name  localhost;
     root   /app/dashboard/;
 
+    gzip on;
+    gzip_comp_level 6;
+    gzip_proxied no-cache no-store private expired auth;
+    gzip_types
+      text/plain
+      text/css
+      application/javascript;
+    gzip_min_length 256;
+
     location /dashboard/ {
         alias /app/dashboard/;
         index  index.html;
```

This reduces the page size by roughly 80% to about 3MB.

### Environment

Browser and version: Firefox 116
OS and version: Linux 6.1
Dashboard version: 3.15.2
Core version: 3.15.4

**Comment by aniav** [comment]
That is true, I checked fro 3.20 and it's sending over 12MB of data. Vendor file contributes 10MB of that. I wonder if we could do something to minify this.

**Comment by rrrnld** [comment]
I already added a patch in the original post that reduces the amount of transferred JS by enabling compression. A better solution would be to additionally minify the JS output when building the dashboard, but given that the patch literally gets you 80% there I guess it's pretty solid already.
